snapcraft/internal/pluginhandler.py

```diff
@@ -656,15 +656,22 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
     for directory in snap_dirs:
         src = os.path.join(srcdir, directory)
         dst = os.path.join(dstdir, directory)
+
+        dir_stat = os.stat(src)
+        uid = dir_stat.st_uid
+        gid = dir_stat.st_gid
         os.makedirs(dst, exist_ok=True)
-        shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
+        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
```

Comment:
So now instead of copying the mode we're copying the owner info. Which is fine. But is there any chance we want to do both?